cms/utils/log_operations.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+
+from django.contrib.admin.models import LogEntry, ADDITION, CHANGE, DELETION
+from django.contrib.contenttypes.models import ContentType
+from django.utils.translation import ugettext_lazy as _
+
+
+"""
+Log helper
+"""
+
+
+def create_log(user_id, content_type_id, object_id, object_repr, action_flag, change_message):
+    """
+    Although this function is repetitive it allows external plugins to access the same logging mechanisms as the cms
+    It also removes the dependency of the admin LogEntry in any other Django CMS code
+    """
+    return LogEntry.objects.log_action(
+        user_id=user_id,
+        content_type_id=content_type_id,
+        object_id=object_id,
+        object_repr=object_repr,
+        action_flag=action_flag,
+        change_message=change_message,
+    )
+
+"""
+Page logs
+"""
+
+
```

Comment:
lots of repetition with the page log functions, also I don't think we should have one for each type of entry.
With the signal handler approach I suggested above, maybe create a page specific handler which receives the operation type and then map operation types to action flags.